src/xbot/common/subsystems/BaseDriveSubsystem.java

```diff
@@ -0,0 +1,142 @@
+package xbot.common.subsystems;
+
+import java.util.Map;
+import com.ctre.CANTalon.TalonControlMode;
+
+import xbot.common.command.BaseSubsystem;
+import xbot.common.controls.actuators.XCANTalon;
+import xbot.common.logging.LoggingLatch;
+import xbot.common.logic.Latch.EdgeType;
+import xbot.common.math.XYPair;
+
+public abstract class BaseDriveSubsystem extends BaseSubsystem {
+    
+    private final LoggingLatch baseDriveSubsystemLoggingLatch = 
+            new LoggingLatch(this.getName(), "XCanTalon(s) in DriveSubsystem is null", EdgeType.RisingEdge);
+    
+    public class MotionRegistration {
+
+        public double x;
+        public double y;
+        public double w;
+        
+        public MotionRegistration(double x, double y, double w) {
+            this.x = x;
+            this.y = y;
+            this.w = w;
+        }        
+        
+        public double calculateTotalImpact(double x, double y, double w) {
+            return this.x * x + this.y*y + this.w * w;
```

Comment:
what's this for? The values are different units, so it seems like this would be useless...

Reply:
I'll make this more clear in some comments - these are actually all in the same "unitless" units: -100% to 100%. 

As an example, let's look at at the front left mecanum wheel. It would be configured as follows:
* X response: 1
* Y response: 1
* Rotation response: -1

If it's then commanded to go forward, strafe right, and rotate right, it will attempt to spin at 300% power.

(We know this isn't possible, which is why we can optionally choose to normalize the outputs if we want to avoid saturating the controls).